shared/profile/routes.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import {Routes} from '../route-tree'
+import Profile from './container'
+
+const userProfileSubTree = {
+  component: Profile,
+  recursive: true,
+  children: {
+    profile: {
+      component: Profile,
```

Comment:
I would think a recursive route would just make the children -> profile be the same as userProfuleSubTree.

i.e. something like this: (this won't work)

```
const userProfileSubTree = {
   component: Profile,
   recursive: true,
   children: {
     profile: userProfileSubTree,
   },
}
```